packages/admin-ui/ui/src/components/molecules/item-search/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { AdminGetInventoryItemsParams, ProductVariant } from "@medusajs/medusa"
+import { ControlProps, OptionProps, SingleValue } from "react-select"
+import { InventoryItemDTO, InventoryLevelDTO } from "@medusajs/types"
+
+import Control from "../select/next-select/components/control"
+import { NextSelect } from "../select/next-select"
+import React from "react"
+
+import SearchIcon from "../../fundamentals/icons/search-icon"
+import { useAdminInventoryItems } from "medusa-react"
+import { useDebounce } from "../../../hooks/use-debounce"
+import { useState } from "react"
+
+type Props = {
+  onItemSelect: (item: itemType) => void
+  clearOnSelect?: boolean
+  filters?: AdminGetInventoryItemsParams
+}
+
+type itemType = Partial<InventoryItemDTO> & {
+  location_levels?: InventoryLevelDTO[] | undefined
+  variants?: ProductVariant[] | undefined
+}
```

Comment:
nit: If the property is optional, an explicit undefined value isn't necessary

```suggestion
type itemType = Partial<InventoryItemDTO> & {
  location_levels?: InventoryLevelDTO[]
  variants?: ProductVariant[]
}
```